src/winrpath.cxx

```diff
@@ -0,0 +1,479 @@
+    PIMAGE_IMPORT_DESCRIPTOR import_image_descriptor = (PIMAGE_IMPORT_DESCRIPTOR)(import_table_offset);
+    //DLL Imports
+    for (; import_image_descriptor->Name != 0; import_image_descriptor++) {
+        DWORD Imported_DLL = import_table_offset + (import_image_descriptor->Name - RVA_import_directory);
+        std::ostringstream str_stream;
+        str_stream << Imported_DLL;
+        if(this->spack_check_for_dll(str_stream.str())) {
+            if(!this->rename_dll(Imported_DLL, str_stream.str())) {
+                throw SpackException("Unable to relocate DLL");
+            }
+        }
+    }
+    if(!CloseHandle(basepointer)){
+        std::cerr << "Unable to properly close file view\n";
+        return -3;
+    }
+    if(!CloseHandle(hMapObject)){
+        std::cerr << "Unable to properly close file map\n";
+        return -4;
+    }
+}
+
+LibRename::LibRename(std::string lib, bool full, bool deploy, bool replace)
+: replace(replace), full(full), lib(lib), deploy(deploy)
+{
+    this->name = stem(this->lib);
+    this->def_file = this->name + ".def";
+    this->def_executor = ExecuteCommand("dumpbin.exe", {this->compute_def_line()});
+    this->lib_executor = ExecuteCommand("lib.exe", {this->compute_rename_line()});
+}
+
+std::string LibRename::compute_def_line()
+{
+    return "/EXPORTS " + this->lib;
+}
+
+void LibRename::computeDefFile()
+{
+    this->def_executor.execute(this->def_file);
+}
+
+int LibRename::executeRename()
+{
+    try {
+        if(!this->deploy){
+            this->computeDefFile();
+            this->executeLibRename();
+        }
+        if (this->full) {
+            this->executeDllRename();
+        }
+    }
+    catch (SpackException &e) {
+        std::cerr << e.what() << "\n";
+        return -1;
+    }
+    return 0;
+}
+
+void LibRename::executeLibRename()
+{
+    this->lib_executor.execute();
+    // import library has been generated with
+    // mangled abs path to dll -
+    // unmangle it
+    CoffReader cr(this->new_lib);
+    CoffParser coff(&cr);
+    coff.parse();
+    coff.normalize_name();
+}
+
+
+void LibRename::executeDllRename()
+{
+    LPCWSTR lib_name = ConvertAnsiToWide(this->lib).c_str();
+    HANDLE dll_handle = CreateFileW(lib_name, (GENERIC_READ|GENERIC_WRITE), FILE_SHARE_READ, NULL, OPEN_EXISTING, FILE_ATTRIBUTE_NORMAL, NULL);
+    if (!dll_handle){
+        std::stringstream os_error;
+        os_error << GetLastError();
+        throw SpackException(os_error.str());
+    }
+    this->find_dll_and_rename(dll_handle);
+}
+
+std::string LibRename::compute_rename_line()
+{
+    std::string line("-def:");
+    line += this->def_file + " ";
+    line += "-name:";
+    line += mangle_name(this->lib) + " ";
+    std::string name(stem(this->lib));
+    if (!this->replace){
+        this->new_lib = name + ".abs-name.lib";
+    }
+    else {
+        this->new_lib = this->lib;
+    }
+    line += "-out:\""+ this->new_lib + "\"" + " " + this->lib;
```

Comment:
if `new_lib` and `lib` have the same name, we're overwriting `lib`, if not, we're creating a second import lib named `new_lib` adjacent to `lib`. I thought this would be useful for when we create the "rpath"-d import libraries for externals, as we should keep the original import libs around for other (non Spack) consumers, but for Spack derived packages we're ok to just replace the import libraries and cut down on our disk memory footprint.